src/utilities/phone/index.ts

```diff
@@ -18,20 +18,56 @@ export function isValidDDD(phone: string): boolean {
   return VALID_AREA_CODES.includes(Number(phone.substr(0, 2)));
 }
 
+export function isValidMobilePhoneLength(phone: string): boolean {
+  return phone.length === PHONE_MAX_LENGTH;
+}
+
+export function isValidLandlinePhoneLength(phone: string): boolean {
+  return phone.length >= PHONE_MIN_LENGTH && phone.length < PHONE_MAX_LENGTH;
+}
+
 export function isValidLength(phone: string): boolean {
-  return phone.length >= PHONE_MIN_LENGTH && phone.length <= PHONE_MAX_LENGTH;
+  return isValidLandlinePhoneLength(phone) || isValidMobilePhoneLength(phone);
+}
+
+export function isValidMobilePhoneFirstNumber(phone: string): boolean {
+  return MOBILE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)));
+}
+
+export function isValidLandlinePhoneFirstNumber(phone: string): boolean {
+  return LANDLINE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)));
 }
 
 export function isValidFirstNumber(phone: string): boolean {
   return phone.length === PHONE_MIN_LENGTH
-    ? LANDLINE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)))
-    : MOBILE_VALID_FIRST_NUMBERS.includes(Number(phone.charAt(2)));
+    ? isValidLandlinePhoneFirstNumber(phone)
+    : isValidMobilePhoneFirstNumber(phone);
+}
+
+function tryPhoneDigits(phone: string): { isValidDigits: boolean; digits: string } {
```

Comment:
Maybe `parsePhoneDigits` is a better function name.

Reply:
Indeed. I'm going to change it.